frontend/src/components/nodes/Nodes.tsx

```diff
@@ -21,6 +21,12 @@ export default class extends React.Component<Props> {
     return await new NodesApi().getNodes(count, this.props.role, endTimestamp);
   };
 
+  componentDidUpdate(prevProps: any) {
+    if (this.props.role !== prevProps.role) {
+      this.autoRefreshNodes = autoRefreshHandler(Nodes, this.config);
+    }
+  }
```

Comment:
It is a band-aid. autoRefreshHandler should do this automatically on props changes.

Reply:
it supposed to do that. But I think the function name and parameter is not changed when it goes into the autoRefreshHandler, it won't trigger the update.